DashAI/back/dataloaders/classes/dashai_dataset.py

```diff
@@ -217,6 +226,169 @@ def save_dataset(datasetdict: DatasetDict, path: str) -> None:
         )
 
 
+@beartype
+def check_split_values(
+    train_size: float,
+    test_size: float,
+    val_size: float,
+) -> None:
+    if train_size < 0 or train_size > 1:
+        raise ValueError(
+            "train_size should be in the (0, 1) range "
+            f"(0 and 1 not included), got {val_size}"
+        )
+    if test_size < 0 or test_size > 1:
+        raise ValueError(
+            "test_size should be in the (0, 1) range "
+            f"(0 and 1 not included), got {val_size}"
+        )
+    if val_size < 0 or val_size > 1:
+        raise ValueError(
+            "val_size should be in the (0, 1) range "
+            f"(0 and 1 not included), got {val_size}"
+        )
+
+
+@beartype
+def split_indices(
+    total_rows: int,
+    train_size: float,
+    test_size: float,
+    val_size: float,
+    seed: Union[int, None] = None,
+    shuffle: bool = True,
+) -> Tuple:
```

Comment:
```suggestion
) -> Tuple[List, List, List]:
```

I take advantage of this same space to ask, why not return the np.array instead of the lists?

I think that conversion could be avoided without problems.

Reply:
Hi! I did the conversion because theres other cases where split_dataset takes List and not np.array, but now I removed the conversion and instaed added the NDArray type to the parameters of split_dataset, you can check this on the recent "remove np.array to list conversion on split_indexes function" commit.

Reply:
I just checked that this change throws an error using python 3.8, I don't have this issue with python 3.10

Reply:
What's the error?

Reply:
I can't include List and NDArray with the operand | since these are both "Generic Alias" types, you can check the error on the failed pytest check.

This is the only issue in Github I have found by the moment: https://github.com/python/cpython/issues/86399, so if we keep working with python 3.8 and 3.9 I think I will have to convert one of the cases to only accept one of the types in the function.

Reply:
Ah ok, let's leave it at that then :)